### ISSUE TITLE
Make easier to extend.

### ISSUE THREAD

**Issue by marcelopm** [hamburgscleanest/guzzle-advanced-throttle#131]
First of all, congrats on the project. Really nice idea and seems very powerful.

One thing I noticed is that private class attributes seems to be use broadly.
I'd recommend avoiding using them unless it's really necessary, as it makes the project much less extendable, resulting in unavoidable dup of code in order to do so.

e.g. Imagine I want create and use a new Driver, that writes cache to DB, as Laravel does out of the box.

I'd have to create a new CacheConfigHelper class from scratch as its DRIVERS constant is private:

https://github.com/hamburgscleanest/guzzle-advanced-throttle/blob/master/src/Cache/Helpers/CacheConfigHelper.php#L23

I them would have to create a new LaravelAdapter from scratch too, because $_cacheManager is also private:

https://github.com/hamburgscleanest/guzzle-advanced-throttle/blob/master/src/Cache/Adapters/LaravelAdapter.php#L23

And so on..

This is just an example, but it might apply to other parts too.

**Comment by lupinitylabs** [comment]
Agreed. It's always a good idea to use `protected` instead of `private` access unless you really don't want derived classes to be able to override anything for a good reason.
